app/src/main/java/com/github/se/eduverse/ui/folder/DisplayFolder.kt

```diff
@@ -112,157 +113,152 @@ fun FolderScreen(
+                                folderViewModel.sortBy(FilterTypes.ACCESS_MOST)
+                                sorting = false
+                              })
+                          DropdownMenuItem(
+                              text = { Text("Least accessed", modifier = Modifier.fillMaxWidth()) },
+                              modifier = Modifier.fillMaxWidth().testTag("sortMode7"),
+                              onClick = {
+                                folderViewModel.sortBy(FilterTypes.ACCESS_LEAST)
+                                sorting = false
+                              })
+                        }
+                  }
                 }
-              }
 
-          // The files
-          activeFolder!!.files.forEach {
-            Button(
-                onClick = { fileViewModel.openFile(it.fileId, context) },
-                modifier =
-                    Modifier.fillMaxWidth()
-                        .padding(horizontal = 20.dp)
-                        .align(Alignment.Start)
-                        .testTag(it.name)) {
-                  Box(
-                      modifier = Modifier.fillMaxWidth(),
-                      contentAlignment = Alignment.CenterStart) {
-                        Text(it.name)
-                        IconButton(
-                            onClick = {
-                              suppressFile = it
-                              dialogOpen = true
-                            },
-                            modifier =
-                                Modifier.align(Alignment.TopEnd)
-                                    .testTag("delete_icon_${it.name}")) {
-                              Icon(Icons.Default.Close, contentDescription = "Delete File")
-                            }
-                      }
-                }
+            // The files
+            activeFolder!!.files.forEach {
+              @Suppress("UNUSED_EXPRESSION")
+              trigger // Force recomposition by adding dependency on a state flow
+
+              Button(
+                  onClick = { fileViewModel.openFile(it.fileId, context) },
+                  modifier = Modifier.fillMaxWidth().padding(20.dp, 3.dp).testTag(it.name)) {
+                    Box(
+                        modifier = Modifier.fillMaxWidth(),
+                        contentAlignment = Alignment.CenterStart) {
+                          Text(it.name)
+                          EditFileMenu(
+                              modifier = Modifier.align(Alignment.CenterEnd),
+                              onRename = {
+                                modifiedFile = it
+                                renameDialogOpen = true
+                              },
+                              onDelete = {
+                                modifiedFile = it
+                                deleteDialogOpen = true
+                              })
+                        }
+                  }
+            }
           }
```

Comment:
as said in the feedback we should stop hard coding the colors and start using the theme file for example.